lib/llm/src/model_card/model.rs

```diff
@@ -402,36 +396,123 @@ struct HFConfig {
     model_type: String,
 
     text_config: Option<HFTextConfig>,
+
+    // Sometimes it's inside HFTextConfig, sometimes it's here
+    eos_token_id: Option<serde_json::Value>,
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 struct HFTextConfig {
-    bos_token_id: TokenIdType,
+    // It can take multiple attempts to load this, so Option
+    bos_token_id: Option<TokenIdType>,
+
+    // We set this once bos_token_id is loaded so we don't have to deal with Option
+    #[serde(default)]
+    final_bos_token_id: TokenIdType,
 
-    #[serde(with = "either::serde_untagged")]
-    eos_token_id: Either<TokenIdType, Vec<TokenIdType>>,
+    eos_token_id: Option<serde_json::Value>,
+
+    #[serde(default)]
+    final_eos_token_ids: Vec<TokenIdType>,
 
     /// max sequence length
-    max_position_embeddings: usize,
+    max_position_embeddings: Option<usize>,
 
     /// number of layers in the model
     num_hidden_layers: usize,
 
     /// number of attention heads in the model
-    num_attention_heads: usize,
+    num_attention_heads: Option<usize>,
 
     /// Vocabulary size
-    vocab_size: usize,
+    vocab_size: Option<usize>,
 }
 
 impl HFConfig {
     async fn from_json_file(file: &str) -> Result<Arc<dyn ModelInfo>> {
+        let file_pathbuf = PathBuf::from(file);
         let contents = std::fs::read_to_string(file)?;
         let mut config: Self = serde_json::from_str(&contents)?;
         if config.text_config.is_none() {
             let text_config: HFTextConfig = serde_json::from_str(&contents)?;
             config.text_config = Some(text_config);
         }
+        // Sometimes bos_token_id is in generation_config.json not config.json
+        let Some(text_config) = config.text_config.as_mut() else {
```

Comment:
This is looking to get tough to maintain - I feel your pain on the inconsistency on model/HF side.